src/main/java/edu/ucsb/ucsbcslas/controllers/TutorAssignmentController.java

```diff
@@ -221,55 +235,109 @@
+        }
+        newAssignment.setAssignmentType(ta.getString("assignmentType"));
 
+        newAssignment.setId(id);
+        tutorAssignmentRepository.save(newAssignment);
+        String body = mapper.writeValueAsString(newAssignment);
+        return ResponseEntity.ok().body(body);
+    }
 
-      newAssignment.setId(id);
-      tutorAssignmentRepository.save(newAssignment);
-      String body = mapper.writeValueAsString(newAssignment);
-      return ResponseEntity.ok().body(body);
+    // Controller to upload CSV for tutorAssignment Service
+
+    @PostMapping(value = "/api/member/tutorAssignments/upload", produces = "application/json")
+    public ResponseEntity<String> uploadCSV(@RequestParam("csv") MultipartFile csv,
+            @RequestHeader("Authorization") String authorization) throws IOException {
+        logger.info("Starting upload CSV");
+        AppUser user = authControllerAdvice.getUser(authorization);
+        try {
+            Reader reader = new InputStreamReader(csv.getInputStream());
+            logger.info(new String(csv.getInputStream().readAllBytes()));
+            List<TutorAssignmentModel> uploadedTutorAssignments = csvToObjectService.parse(reader,
+                    TutorAssignmentModel.class);
+            List<TutorAssignment> tutorAssignmentList = new ArrayList<TutorAssignment>();
+
+            for (TutorAssignmentModel i : uploadedTutorAssignments) {
+                Course currentCourse;
+                Tutor currentTutor;
+                TutorAssignment currentTutorAssignment;
+
+                List<Course> courseList = courseRepository.findByNameAndQuarterAndInstructorEmail(i.getCourseName(),
+                        i.getQuarter(), i.getInstructorEmail());
+                if (courseList.isEmpty()) {
+                    currentCourse = new Course(i.getCourseName(), i.getQuarter(), i.getInstructorFirstName(),
+                            i.getInstructorLastName(), i.getInstructorEmail());
+                    courseRepository.save(currentCourse);
+                } else {
+                    currentCourse = courseList.get(0);
+                }
+                Optional<Tutor> tutorList = tutorRepository.findByEmail(i.getTutorEmail());
+                if (tutorList.isEmpty()) {
+                    currentTutor = new Tutor(i.getTutorFirstName(), i.getTutorLastName(), i.getTutorEmail());
+                    tutorRepository.save(currentTutor);
+                } else {
+                    currentTutor = tutorList.get();
+                }
+                tutorAssignmentList = tutorAssignmentRepository.findAllByTutor(currentTutor);
+                if (tutorAssignmentList.isEmpty()) {
+                    currentTutorAssignment = new TutorAssignment(currentCourse, currentTutor, i.getAssignmentType());
+                    tutorAssignmentRepository.save(currentTutorAssignment);
+                } else {
+                    currentTutorAssignment = tutorAssignmentList.get(0);
+                }
```

Comment:
I think this else statement can be removed, as `currentTutorAssignment` is not used after

Reply:
Good catch!